test/replication-luatest/gh_7253_election_long_wal_write_test.lua

```diff
@@ -0,0 +1,173 @@
+    box.error.injection.set('ERRINJ_WAL_DELAY_COUNTDOWN', 0)
+end
+
+local function unblock_wal_write_f()
+    box.error.injection.set('ERRINJ_WAL_DELAY', false)
+end
+
+local function server_block_next_wal_write(server)
+    server:exec(block_next_wal_write_f)
+end
+
+local function server_unblock_wal_write(server)
+    server:exec(unblock_wal_write_f)
+end
+
+local function check_wal_is_blocked_f()
+    if not box.error.injection.get('ERRINJ_WAL_DELAY') then
+        error('WAL still is not paused')
+    end
+end
+
+local function server_wait_wal_is_blocked(server)
+    t.helpers.retrying({timeout = wait_timeout}, server.exec, server,
+                       check_wal_is_blocked_f)
+end
+
+g.before_all(function(g)
+    g.cluster = cluster:new({})
+    local box_cfg = {
+        replication_synchro_timeout = 1000,
+        replication_synchro_quorum = 2,
+        replication_timeout = 0.1,
+        replication = {
+            server.build_instance_uri('server1'),
+            server.build_instance_uri('server2'),
+            server.build_instance_uri('server3'),
+        },
+    }
+    box_cfg.listen = box_cfg.replication[1]
+    box_cfg.election_mode = 'manual'
+    g.server1 = g.cluster:build_and_add_server({
+        alias = 'server1', box_cfg = box_cfg
+    })
+    box_cfg.listen = box_cfg.replication[2]
+    box_cfg.election_mode = 'voter'
+    g.server2 = g.cluster:build_and_add_server({
+        alias = 'server2', box_cfg = box_cfg
+    })
+    box_cfg.listen = box_cfg.replication[3]
+    g.server3 = g.cluster:build_and_add_server({
+        alias = 'server3', box_cfg = box_cfg
+    })
+    g.cluster:start()
+
+    g.server1:exec(function()
+        local s = box.schema.create_space('test', {is_sync = true})
+        s:create_index('pk')
+    end)
+    wait_fullmesh(g)
+end)
+
+g.after_all(function(g)
+    g.cluster:drop()
+    g.server1 = nil
+    g.server2 = nil
+    g.server3 = nil
+end)
+
+g.test_fence_during_confirm_wal_write = function(g)
+    --
+    -- Make the following topology:
+    --   server2 <-> server1 <- server3
+    --
+    g.server2:exec(function()
+        local replication = table.copy(box.cfg.replication)
+        rawset(_G, 'old_replication', table.copy(replication))
+        replication[3] = nil
+        box.cfg{replication = replication}
+    end)
+    g.server3:exec(function()
+        rawset(_G, 'old_replication', table.copy(box.cfg.replication))
+        box.cfg{replication = {}}
+    end)
+    --
+    -- Server1 starts a txn.
+    --
+    server_block_next_wal_write(g.server2)
+    local f = fiber.new(g.server1.exec, g.server1, function()
+        box.space.test:replace({1})
+    end)
+    f:set_joinable(true)
+    --
+    -- The txn is delivered to server2.
+    --
+    server_wait_wal_is_blocked(g.server2)
+    --
+    -- Server1 receives an ack and blocks on CONFIRM WAL write.
+    --
+    server_block_next_wal_write(g.server1)
+    server_unblock_wal_write(g.server2)
```

Comment:
It seems you could test this with only two servers:
servers 1 and 2 would act the same, and server 2 would bump the term instead of server3.

Reply:
Indeed, thanks for the comment, done.